### DISCUSSION TITLE
Usage of theia as Computer Assisted Translation Software (CATS) for The Free Software Foundation Europe?

### DISCUSSION THREAD

**Discussion by Kreyren** [eclipse-theia/theia#9026] [Q&A] [answered]
## Abstract

Making is easier for translators and proof-readers to translate the FSFE websites.

## Practical Abstract

Method to process XML files in a user-friendly way and submit them for review with option to go back and forth until the proposed change is up to the standard.

## Context

Web team was today 04/02/2021 discussing the implementation of provided abstract where it was decided to implement custom solution https://git.fsfe.org/fsfe-system-hackers/webpreview which alpha version is currently provided on https://webpreview.fsfe.org.

## Question/Issue

My personal opinion is that the proposed new features added to the webpreview make it more and more into a custom IDE so i was wondering if theia could an alternative solution there?

The projected solution would be basically:
```
__________________________________________________________________
| File browser        | Text to translate          | Preview      |
|                     |                            |              |
|                     |                            |              |
|                     |                            |              |
|                     |                            |              |
|                     |                            |              |
__________________________________________________________________
```

Where ideally theia should translate the codeblocks on demand so that the translator could just verify the translate and correct it e.g. how crowdin is doing this.

![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fd2.alternativeto.net%2Fdist%2Fs%2Fcrowdin_530395_full.png%3Fformat%3Djpg%26width%3D1600%26height%3D1600%26mode%3Dmin%26upscale%3Dfalse&f=1&nofb=1)

**Comment by vince-fugnitto** [comment, marked as answer]
@Kreyren thank you for the discussion.

If the question is whether the framework can be used to implement such an application then I don't see why not (given that you implement the application using custom theia extensions, and possibly customize the default behavior of the framework if needed). You can even possibly create a language-server for translating such documents that your application, theia, vscode (or any app supporting LSP) can spawn and consume.

Sorry if we cannot give you more information or guidance.

**Reply by Kreyren** [reply]
Added that on my TODO list then unless this extension already exists?

So far i was looking into Crowdin which works by submitting the translations from the editor to the API where it is then processed for merging, but it is not user-friendly enough (Seems usable for developers though).

**Reply by vince-fugnitto** [reply]
> Added that on my TODO list then unless this extension already exists?

I'm not aware of any theia extension which exists that contributes such functionality, but you can research any tools which perform the translation.
